homeassistant/components/neato/config_flow.py

```diff
@@ -1,112 +1,60 @@
-"""Config flow to configure Neato integration."""
-
+"""Config flow for Neato Botvac."""
 import logging
-
-from pybotvac import Account, Neato, Vorwerk
-from pybotvac.exceptions import NeatoLoginException, NeatoRobotException
-import voluptuous as vol
+from typing import Optional
 
 from homeassistant import config_entries
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers import config_entry_oauth2_flow
 
 # pylint: disable=unused-import
-from .const import CONF_VENDOR, NEATO_DOMAIN, VALID_VENDORS
+from .const import NEATO_DOMAIN
 
 DOCS_URL = "https://www.home-assistant.io/integrations/neato"
-DEFAULT_VENDOR = "neato"
 
 _LOGGER = logging.getLogger(__name__)
 
 
-class NeatoConfigFlow(config_entries.ConfigFlow, domain=NEATO_DOMAIN):
-    """Neato integration config flow."""
+class OAuth2FlowHandler(
+    config_entry_oauth2_flow.AbstractOAuth2FlowHandler, domain=NEATO_DOMAIN
+):
+    """Config flow to handle Neato Botvac OAuth2 authentication."""
 
-    VERSION = 1
+    VERSION = 2
+    DOMAIN = NEATO_DOMAIN
     CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
 
-    def __init__(self):
-        """Initialize flow."""
-        self._username = vol.UNDEFINED
-        self._password = vol.UNDEFINED
-        self._vendor = vol.UNDEFINED
-
-    async def async_step_user(self, user_input=None):
-        """Handle a flow initialized by the user."""
-        errors = {}
-
-        if self._async_current_entries():
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
+
+    async def async_step_user(
+        self, user_input: Optional[dict] = None, reauth: bool = False
+    ) -> dict:
+        """Create an entry for the flow."""
+        if reauth:
+            # Ongoig migration
+            return self.async_show_form(step_id="user")
```

Comment:
I'd probably have a dedicated form in the reauth step or a confirm step instead.

Reply:
I think it would be easier to see what's going on if we have a dedicated form in the reauth or reauth confirm step.

Reply:
Yea, I'll do that. Or I'll try :smile:

Reply:
Done... I guess...